File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon_species/FlavourEntries.java
package pt.rfsfernandes.model.pokemon_species;

import com.google.gson.annotations.SerializedName;

import androidx.room.TypeConverters;
import pt.rfsfernandes.data.local.PokemonTypeConverters;
import pt.rfsfernandes.model.SimpleModelData;

/**
 * Class FlavourEntries created at 1/17/21 20:02 for the project PoketDex
 * By: rodrigofernandes
 */
@TypeConverters(PokemonTypeConverters.class)
public class FlavourEntries {
  @SerializedName("flavor_text")
  private String flavourText;

  @SerializedName("language")
  private SimpleModelData language;

  public FlavourEntries(String flavourText, SimpleModelData language) {
    this.flavourText = flavourText;
    this.language = language;
  }

  public SimpleModelData getLanguage() {
    return language;
  }

  public void setLanguage(SimpleModelData language) {
    this.language = language;
  }

  public String getFlavourText() {
    return flavourText.replace("\n", " ");
  }

  public void setFlavourText(String flavourText) {
    this.flavourText = flavourText;
  }
}

File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon_species/FlavourEntries.kt
package pt.rfsfernandes.model.pokemon_species

import androidx.room.TypeConverters
import pt.rfsfernandes.data.local.PokemonTypeConverters
import pt.rfsfernandes.model.SimpleModelData
import com.google.gson.annotations.SerializedName
import pt.rfsfernandes.data.local.PokemonSpeciesTypeConverters
import pt.rfsfernandes.model.pokemon_species.FlavourEntries
import androidx.room.PrimaryKey

/**
 * Class FlavourEntries created at 1/17/21 20:02 for the project PoketDex
 * By: rodrigofernandes
 */
@TypeConverters(PokemonTypeConverters::class)
class FlavourEntries(@field:SerializedName("flavor_text") private var flavourText: String, @field:SerializedName("language") var language: SimpleModelData) {
    fun getFlavourText(): String {
        return flavourText.replace("\n", " ")
    }

    fun setFlavourText(flavourText: String) {
        this.flavourText = flavourText
    }
}